lib/rules/indent.js

```diff
@@ -807,6 +889,84 @@ module.exports = {
             });
         }
 
+        /**
+         * Iterate through the dependent tokens until we reach the last token
+         * dependency or the last token dependency that is within the unknown
+         * node. If the last token dependency is not the first token of the line
+         * make the last token dependcy depend on the first token of the line.
+         * This ensures the offset will be set correctly since only the first
+         * token of each line is ignored and not all tokens on a line.
+         * @param {Token} dependency First
+         * @param {number} startLineOfNode First line number of unknown node
+         * @returns {void}
+         */
+        function ignoreUnknownNodeLineWithDependency(dependency, startLineOfNode) {
+            let currDependency = dependency,
+                lastDependency = dependency;
+
+            while (
+                (currDependency = offsets.getFirstDependency(currDependency)) &&
+                currDependency.loc.start.line > startLineOfNode
```

Comment:
Feel free to push the working solution. It's possible there's a better way, but this one might be fine for now. (We can always improve the solution in the future -- I mainly want to make sure we wouldn't be breaking backwards compatibility by improving the solution later, e.g. if we're ignoring too many tokens here.)